jdaviz/configs/imviz/plugins/coords_info/coords_info.py

```diff
@@ -384,10 +391,17 @@ def _image_viewer_update(self, viewer, x, y, mouseevent=True):
             else:
                 data_wcs = None
 
+            if "spectral_axis_index" in getattr(coo_data, "meta", {}):
+                self._spectral_axis_index = coo_data.meta["spectral_axis_index"]
```

Comment:
I see this is grabbing from the data... do we need to set it as a class attribute at all then?

Reply:
It does get used in one or two other places, it seemed more convenient to store it rather than access the data meta every time.